<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Ethernet</class>
 <widget class="QWidget" name="Ethernet">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>661</width>
    <height>731</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>641</width>
        <height>711</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="ethernet_hostname_label">
           <property name="text">
            <string>Hostname:</string>
           </property>
           <property name="buddy">
            <cstring>ethernet_hostname</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QLineEdit" name="ethernet_hostname">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
             <horstretch>1</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maxLength">
            <number>32</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Socket Port:</string>
           </property>
           <property name="buddy">
            <cstring>ethernet_port</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <item>
            <widget class="QSpinBox" name="ethernet_port">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>65535</number>
             </property>
             <property name="value">
              <number>4223</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_6">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>(Plain) /</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="ethernet_websocket_port">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>65535</number>
             </property>
             <property name="value">
              <number>4280</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>(Web)</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>Max Connections:</string>
           </property>
           <property name="buddy">
            <cstring>ethernet_socket_connections</cstring>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_11">
           <item>
            <widget class="QSpinBox" name="ethernet_socket_connections">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>7</number>
             </property>
             <property name="value">
              <number>4</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_5">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>(Plain) /</string>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="ethernet_websocket_connections">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>7</number>
             </property>
             <property name="value">
              <number>3</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>(Web)</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="3" column="1">
          <widget class="QCheckBox" name="ethernet_use_auth">
           <property name="text">
            <string>Use Authentication</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_12">
           <item>
            <widget class="QLineEdit" name="ethernet_secret">
             <property name="maxLength">
              <number>64</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="ethernet_show_characters">
             <property name="text">
              <string>Show Characters</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="ethernet_secret_label">
           <property name="text">
            <string>Secret:</string>
           </property>
           <property name="buddy">
            <cstring>ethernet_secret</cstring>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="label_7">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Address:</string>
           </property>
           <property name="buddy">
            <cstring>ethernet_connection</cstring>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QComboBox" name="ethernet_connection">
           <item>
            <property name="text">
             <string>DHCP</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Static IP</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="7" column="0">
          <widget class="QLabel" name="ethernet_ip_label">
           <property name="text">
            <string>IP:</string>
           </property>
           <property name="buddy">
            <cstring>ethernet_ip4</cstring>
           </property>
          </widget>
         </item>
         <item row="8" column="0">
          <widget class="QLabel" name="ethernet_sub_label">
           <property name="text">
            <string>Subnet Mask:</string>
           </property>
           <property name="buddy">
            <cstring>ethernet_sub4</cstring>
           </property>
          </widget>
         </item>
         <item row="9" column="0">
          <widget class="QLabel" name="ethernet_gw_label">
           <property name="text">
            <string>Gateway:</string>
           </property>
           <property name="buddy">
            <cstring>ethernet_gw4</cstring>
           </property>
          </widget>
         </item>
         <item row="7" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_13">
           <item>
            <widget class="QSpinBox" name="ethernet_ip4">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="ethernet_dot1">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>.</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="ethernet_ip3">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="ethernet_dot2">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>.</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="ethernet_ip2">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="ethernet_dot3">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>.</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="ethernet_ip1">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="8" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_14">
           <item>
            <widget class="QSpinBox" name="ethernet_sub4">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="ethernet_dot4">
             <property name="text">
              <string>.</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="ethernet_sub3">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="ethernet_dot5">
             <property name="text">
              <string>.</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="ethernet_sub2">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="ethernet_dot6">
             <property name="text">
              <string>.</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="ethernet_sub1">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="9" column="1">
          <layout class="QHBoxLayout" name="horizontalLayout_16">
           <item>
            <widget class="QSpinBox" name="ethernet_gw4">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="ethernet_dot7">
             <property name="text">
              <string>.</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="ethernet_gw3">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="ethernet_dot8">
             <property name="text">
              <string>.</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="ethernet_gw2">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="ethernet_dot9">
             <property name="text">
              <string>.</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="ethernet_gw1">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="11" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>MAC Address:</string>
           </property>
           <property name="buddy">
            <cstring>ethernet_mac6</cstring>
           </property>
          </widget>
         </item>
         <item row="11" column="1">
          <layout class="QHBoxLayout" name="mac_layout">
           <item>
            <widget class="SpinBoxHex" name="ethernet_mac6">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string>:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="SpinBoxHex" name="ethernet_mac5">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="SpinBoxHex" name="ethernet_mac4">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="SpinBoxHex" name="ethernet_mac3">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_14">
             <property name="text">
              <string>:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="SpinBoxHex" name="ethernet_mac2">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_15">
             <property name="text">
              <string>:</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="SpinBoxHex" name="ethernet_mac1">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximum">
              <number>255</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="5" column="0" colspan="2">
          <widget class="Line" name="line_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item row="10" column="0" colspan="2">
          <widget class="Line" name="line">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item row="12" column="0" colspan="2">
          <widget class="Line" name="line_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item row="16" column="0" colspan="2">
          <widget class="QPushButton" name="ethernet_save">
           <property name="text">
            <string>Save Ethernet Configuration</string>
           </property>
          </widget>
         </item>
         <item row="15" column="0" colspan="2">
          <widget class="Line" name="line_5">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item row="13" column="0">
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>RX Count:</string>
           </property>
          </widget>
         </item>
         <item row="13" column="1">
          <widget class="QLabel" name="ethernet_count_rx">
           <property name="text">
            <string>0</string>
           </property>
          </widget>
         </item>
         <item row="14" column="0">
          <widget class="QLabel" name="label_10">
           <property name="text">
            <string>TX Count:</string>
           </property>
          </widget>
         </item>
         <item row="14" column="1">
          <widget class="QLabel" name="ethernet_count_tx">
           <property name="text">
            <string>0</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SpinBoxHex</class>
   <extends>QSpinBox</extends>
   <header location="global">brickv.spin_box_hex</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>scrollArea</tabstop>
  <tabstop>ethernet_hostname</tabstop>
  <tabstop>ethernet_port</tabstop>
  <tabstop>ethernet_websocket_port</tabstop>
  <tabstop>ethernet_socket_connections</tabstop>
  <tabstop>ethernet_websocket_connections</tabstop>
  <tabstop>ethernet_use_auth</tabstop>
  <tabstop>ethernet_secret</tabstop>
  <tabstop>ethernet_show_characters</tabstop>
  <tabstop>ethernet_connection</tabstop>
  <tabstop>ethernet_ip4</tabstop>
  <tabstop>ethernet_ip3</tabstop>
  <tabstop>ethernet_ip2</tabstop>
  <tabstop>ethernet_ip1</tabstop>
  <tabstop>ethernet_sub4</tabstop>
  <tabstop>ethernet_sub3</tabstop>
  <tabstop>ethernet_sub2</tabstop>
  <tabstop>ethernet_sub1</tabstop>
  <tabstop>ethernet_gw4</tabstop>
  <tabstop>ethernet_gw3</tabstop>
  <tabstop>ethernet_gw2</tabstop>
  <tabstop>ethernet_gw1</tabstop>
  <tabstop>ethernet_mac6</tabstop>
  <tabstop>ethernet_mac5</tabstop>
  <tabstop>ethernet_mac4</tabstop>
  <tabstop>ethernet_mac3</tabstop>
  <tabstop>ethernet_mac2</tabstop>
  <tabstop>ethernet_mac1</tabstop>
  <tabstop>ethernet_save</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
